doozer/tests/test_lockfile.py

```
@@ -812,6 +812,69 @@ async def test_generate_artifact_lockfile_success(self, mock_session_class, mock
+    @patch('aiohttp.ClientSession')
+    async def test_generate_artifact_lockfile_custom_filename(self, mock_session_class, mock_write_yaml):
+        """Test artifact lockfile generation with custom destination filenames"""
+        # Setup mocks
+        mock_image_meta = MagicMock()
+        mock_image_meta.distgit_key = "test-image"
+        mock_image_meta.get_required_artifacts.return_value = [
+            {'url': "https://example.com/v1.2.3/long-path/file.jar", 'filename': "custom.jar"},
+            {'url': "https://example.com/cert.pem", 'filename': None},  # Auto-extract
+        ]
+
+        mock_dest_dir = Path("/tmp/test")
```

Ruff (0.14.14)

[error] Probable insecure usage of temporary file or directory: "/tmp/test"

(S108)